packages/txservice/src/provider.ts

```diff
@@ -450,4 +464,85 @@ export class ChainRpcProvider {
   private incrementNonce() {
     this._nonce++;
   }
+
+  private async monitorBlockades() {
+    while (true) {
+      // First, check to see that there are pending txs
+      if (Object.keys(this.pendingTxs).length === 0) {
+        return;
```

Comment:
How would this work? `this.monitorBlockades` would be called in constructor and then immediately return... did you mean to maybe do a `delay(1000);` then `continue;` or something?